### PR TITLE
enable robot full access

### PR DESCRIPTION
**Pull request by wy65701436**
When the system admin enable this option, the robot can be assigned with robot/user/group/quota permissions.

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/common/const.go [hunk 1]
```diff
@@ -246,4 +246,7 @@ const (
 
 	// Global Leeway used for token validation
 	JwtLeeway = 60 * time.Second
+
+	// Global Leeway used for token validation
```
Comment (stonezdj) [inline src/common/const.go:250]: Wrong comment?

File: src/common/rbac/const.go [hunk 2]
```diff
@@ -81,9 +86,89 @@ const (
...
+}
+
+// GetPermissionProvider gives the robot permission provider
+func GetPermissionProvider(ctx context.Context) RobotPermissionProvider {
+	var permissionProvider RobotPermissionProvider
+	permissionProvider = &BaseProvider{}
+	if config.RobotFullAccess(ctx) {
+		permissionProvider = &NolimitProvider{}
+	}
+	return permissionProvider
```
Comment (stonezdj) [inline src/common/rbac/const.go:108]:
```
if config.RobotFullAccess(ctx) {
    return &NolimitProvider{}
}
return &BaseProvider{}
```

File: src/common/rbac/const.go [hunk 3]
```diff
@@ -81,9 +86,89 @@ const (
...
+type RobotPermissionProvider interface {
+	GetPermissions(s scope) []*types.Policy
+}
+
+// GetPermissionProvider gives the robot permission provider
+func GetPermissionProvider(ctx context.Context) RobotPermissionProvider {
+	var permissionProvider RobotPermissionProvider
+	permissionProvider = &BaseProvider{}
+	if config.RobotFullAccess(ctx) {
+		permissionProvider = &NolimitProvider{}
```
Comment (stonezdj) [inline src/common/rbac/const.go:106]: Nolimit -> Unlimit

**Review by stonezdj — APPROVED** [review]

**Review by chlins — APPROVED** [review]
lgtm

**Review by MinerYang — APPROVED** [review]
lgtm

**Pull request merged**